src/scenery.cpp

```diff
@@ -244,19 +246,28 @@ void SceneryInstance::OnAnimate(const int delay)
 {
 	if (this->type->watering_interval > 0) this->last_watered += delay;
 	this->animtime += delay;
-	this->animtime %= (this->NeedsWatering() ? this->type->dry_animation : this->type->main_animation)->GetTotalDuration();
+	this->animtime %= (this->IsDry() ? this->type->dry_animation : this->type->main_animation)->GetTotalDuration();
 	this->MarkDirty();  // Ensure the animation is updated.
 }
 
 /**
  * Whether this item is dried up for lack of watering.
  * @return The item is dry.
  */
-bool SceneryInstance::NeedsWatering() const
+bool SceneryInstance::IsDry() const
 {
 	return this->type->watering_interval > 0 && this->last_watered > this->type->watering_interval;
```

Comment:
Minor thing: Seem to have missed this before, but comparing `last_watered` against an interval looks odd at first sight, since `last_watered` doesn't express time very clearly (it could be a boolean whether this instance got watered last).

Perhaps `time_since_watered`?  (`last` doesn't look very important to me as code generally doesn't track history, it's also somewhat implied with `since`.)